dask/array/core.py

```diff
@@ -1080,6 +1080,16 @@ def numblocks(self):
     def npartitions(self):
         return reduce(mul, self.numblocks, 1)
 
+    def persist(self):
+        return super().persist()
+
+    def compute_metadata(self):
+        from .rechunk import _get_chunks
+
+        chunks = compute(_get_chunks(self))[0]
+        self._chunks = normalize_chunks(chunks, self.shape, dtype=self.dtype)
```

Comment:
Why do we need to call `normalize_chunks` here?  Aren't we getting out the full chunk structure?